tests/dump_tests/module_tests/vlan_test.py

```diff
@@ -0,0 +1,194 @@
+dedicated_dbs['ASIC_DB'] = os.path.join(vlan_files_path, "asic_db.json")
+dedicated_dbs['STATE_DB'] = os.path.join(vlan_files_path, "state_db.json")
+
+def verbosity_setup():
+    print("SETUP")
+    os.environ["VERBOSE"] = "1"
+    yield
+    print("TEARDOWN")
+    os.environ["VERBOSE"] = "0"
+
+def populate_mock(db, db_names):
+    for db_name in db_names:
+        db.connect(db_name)
+        # Delete any default data
+        db.delete_all_by_pattern(db_name, "*")
+        with open(dedicated_dbs[db_name]) as f:
+            mock_json = json.load(f)
+        for key in mock_json:
+            for field, value in mock_json[key].items():
+                db.set(db_name, key, field, value)
+
+@pytest.fixture(scope="class", autouse=True)
+def match_engine():
+
+    # Monkey Patch the SonicV2Connector Object
+    from ...mock_tables import dbconnector
+    db = SonicV2Connector()
+
+    # popualate the db with mock data
+    db_names = list(dedicated_dbs.keys())
+    try:
+        populate_mock(db, db_names)
+    except Exception as e:
+        assert False, "Mock initialization failed: " + str(e)
+
+    # Initialize connection pool
+    conn_pool = ConnectionPool()
+    DEF_NS = ''  # Default Namespace
+    conn_pool.cache = {DEF_NS: {'conn': db,
+                                'connected_to': set(db_names)}}
+
+    # Initialize match_engine
+    match_engine = MatchEngine(conn_pool)
+    yield match_engine
+
+@pytest.mark.usefixtures("match_engine")
+class TestVlanModule:
+    
+    def test_working_state(self, match_engine):
+        params = {}
+        params[Vlan.ARG_NAME] = "Vlan4"
+        params["namespace"] = ""
+        m_vlan = Vlan(match_engine)
+        returned = m_vlan.execute(params)
+        expect = create_template_dict(dbs=["CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_DB"]["keys"].append("VLAN|Vlan4")
+        expect["CONFIG_DB"]["keys"].append("VLAN_MEMBER|Vlan4|Ethernet16")
+        expect["CONFIG_DB"]["keys"].append("VLAN_MEMBER|Vlan4|Ethernet24")
+        expect["APPL_DB"]["keys"].append("VLAN_TABLE:Vlan4")
+        expect["APPL_DB"]["keys"].append("VLAN_MEMBER_TABLE:Vlan4:Ethernet16")
+        expect["APPL_DB"]["keys"].append("VLAN_MEMBER_TABLE:Vlan4:Ethernet24")
+        expect["STATE_DB"]["keys"].append("VLAN_TABLE|Vlan4")
+        expect["STATE_DB"]["keys"].append("VLAN_MEMBER_TABLE|Vlan4|Ethernet16")
+        expect["STATE_DB"]["keys"].append("VLAN_MEMBER_TABLE|Vlan4|Ethernet24")
+        expect["ASIC_DB"]["keys"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN:oid:0x260000000005e5")
+        expect["ASIC_DB"]["keys"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER:oid:0x270000000005e7")
+        expect["ASIC_DB"]["keys"].append("ASIC_STATE:SAI_OBJECT_TYPE_VLAN_MEMBER:oid:0x270000000005e9")
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect), ignore_order=True)
+        assert not ddiff, ddiff
+        
+    def test_missing_interface(self, match_engine):
```

Comment:
Can we remove the interface related tests?

Reply:
Done.